app/controllers/providers/other_assets_controller.rb

```diff
@@ -8,7 +8,8 @@ def show
     def update
       authorize @legal_aid_application
       @form = Citizens::OtherAssetsForm.new(form_params)
-      render :show unless save_continue_or_draft(@form)
+      @form.errors.add :base, :provider_none_selected unless draft_selected? || none_checkbox_selected? || @form.any_checkbox_checked?
+      render :show unless !@form.errors.present? && save_continue_or_draft(@form)
```

Comment:
If the form has errors, I think `save_continue_or_draft(@form)` should fail. So this should be OK:
```ruby
render :show unless save_continue_or_draft(@form)
```

Reply:
Done. In order to do this, I had to move the validation of none_selected to the form and away from the controller.